cmd/syncthing/main.go

```diff
@@ -349,7 +350,12 @@ func main() {
 			// Use leveldb database locks to protect against concurrent upgrades
 			_, err = leveldb.OpenFile(locations[locDatabase], &opt.Options{OpenFilesCacheCapacity: 100})
 			if err != nil {
-				l.Fatalln("Cannot upgrade, database seems to be locked. Is another copy of Syncthing already running?")
+				err = upgradeViaRest()
+				if err != nil {
+					l.Fatalln("The database seems to be locked. Tried to upgrade running Syncthing instance but it failed", err)
```

Comment:
I'd make this shorter, as it will likely not fit on one line now.

```go
if err != nil {
    l.Infoln("Attempting upgrade through running Syncthing...2)
    ...
    if err != nil {
        l.Fatalln("Upgrade:", err)
    }
    l.Okln("Syncthing upgrading")
```

?